modules/concurrency/include/hephaestus/concurrency/spinner.h

```diff
@@ -0,0 +1,34 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <thread>
+
+namespace heph::concurrency {
+
+class Spinner {
+public:
+  Spinner();
+  virtual ~Spinner();
+  Spinner(const Spinner&) = delete;
+  auto operator=(const Spinner&) -> Spinner& = delete;
+  Spinner(Spinner&&) = delete;
+  auto operator=(Spinner&&) -> Spinner& = delete;
+
+  void start();
+  void stop();
+  virtual void spin();
+  virtual void spinOnce() = 0;  // Pure virtual function
+  void addStopCallback(std::function<void()> callback);
+
+private:
+  std::atomic<bool> is_started_ = false;
+  std::atomic<bool> stop_requested_ = false;
```

Comment:
cosmetics:
I don't know if like this is more clean, WDYT?
```suggestion
  std::atomic_bool is_started_ = false;
  std::atomic_bool stop_requested_ = false;
```

Reply:
agree, looks better!